src/test/checkblock_tests.cpp

```diff
@@ -47,18 +56,24 @@ BOOST_FIXTURE_TEST_SUITE(checkblock_tests, BasicTestingSetup) // BU harmonize su
 BOOST_AUTO_TEST_CASE(TestBlock)
 {
     CBlock testblock;
-    if (read_block("testblock.dat", testblock))
+    bool fReadBlock = read_block("testblock.dat", testblock);
+    BOOST_CHECK_MESSAGE(fReadBlock, "Failed to read " + fullFilePath);
+    if (fReadBlock)
     {
         CValidationState state;
 
         uint64_t blockSize = ::GetSerializeSize(testblock, SER_NETWORK, PROTOCOL_VERSION); // 53298 B for test.dat
 
         BOOST_CHECK_MESSAGE(CheckBlock(testblock, state, false, false), "Basic CheckBlock failed");
+        // NOTE: setting of fExcessive was moved from CheckBlock to ContextualCheckBlock in c64d44b7
+        BOOST_CHECK_MESSAGE(ContextualCheckBlock(testblock, state, nullptr, false), "Contextual CheckBlock failed");
```

Comment:
`ContextualCheckBlock` contains function calls which internally assert that `cs_main` is held, so this will fail since the lock isn't being taken in this test.  This appears to be the root cause of the test failures on Travis, though we couldn't see that information until #1836 was added